services/galley/src/Galley/Data.hs

```diff
@@ -687,10 +689,27 @@ privateRole = PrivateAccessRole
 privateOnly :: Set Access
 privateOnly = Set [PrivateAccess]
 
+toMappedOrLocalId :: (Id (Opaque a), Maybe (Id (Remote a)), Maybe Domain) -> MappedOrLocalId a
+toMappedOrLocalId = \case
+  (Id mappedId, Just remoteId, Just domain) ->
+    Mapped (IdMapping (Id mappedId) (Qualified remoteId domain))
+  (Id localId, Nothing, Nothing) ->
+    Local (Id localId)
+  (Id localId, _, _) ->
+    -- TODO(mheinzel): this should be an error
```

Comment:
This basically indicates invalid data in Cassandra, but we need to surface the error somehow.